### PR TITLE
space chats 

### PR DESCRIPTION
**Pull request by gtalha07**
This PR contain additions in relevance to issue #730.

### CODE REVIEW

File: app/lib/features/chat/dialogs/create_chat_sheet.dart [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+                Column(
+                  children: <Widget>[
+                    const Padding(
+                      padding: EdgeInsets.only(bottom: 5),
+                      child: Text('Avatar'),
+                    ),
+                    GestureDetector(
+                      onTap: () => customMsgSnackbar(
+                        context,
+                        'Room Avatar upload isn\'t available yet',
```
[the remark this answers is not in the document]
Comment (gtalha07) [reply app/lib/features/chat/dialogs/create_chat_sheet.dart:77]: Does that mean we are treating ```Conversation``` object as ```Space``` ? We could maybe use ```spaceProvider``` to fetch the space and start configuring via ```CreateSpaceBuilder```. Why would we need ```CreateConversationBuilder``` then?

File: native/core/src/spaces.rs [hunk 2]
```diff
@@ -77,35 +77,49 @@ pub struct CreateSpaceSettings {
...
-        if let Ok(user_id) = UserId::parse(value) {
-            self.invites.push(user_id);
-        }
+    pub fn set_invites(&mut self, values: &mut [String]) -> Result<()> {
+        let invites = values
+            .iter_mut()
+            .filter_map(|x| UserId::parse(x).ok())
+            .collect();
+        self.invites = Some(invites);
+        Ok(())
```
[the remark this answers is not in the document]
Comment (gtalha07) [reply native/core/src/spaces.rs:104]: Perhaps @bitfriend could answer to this...

File: native/acter/api.rsh [hunk 3]
```diff
@@ -1373,7 +1363,56 @@ object PublicSearchResult {
     fn chunks() -> Vec<PublicSearchResultItem>;
 }
 
-fn new_space_settings(name: string, topic: Option<string>, avatar_uri: Option<string>, parent: Option<string>) -> Result<CreateSpaceSettings>;
+/// make convo settings builder
+fn new_convo_settings_builder() -> CreateConversationSettingsBuilder;
+
+object CreateConversationSettingsBuilder {
+    /// set the name of convo
```
[the remark this answers is not in the document]
Comment (gtalha07) [reply native/acter/api.rsh:1370]: I will open issue to that and fix per commit in separate PR.

File: app/lib/features/spaces/dialogs/create_space_sheet.dart [hunk 4]
```diff
@@ -290,14 +290,19 @@ class _CreateSpacePageConsumerState extends ConsumerState<CreateSpacePage> {
...
-      parentRoomId,
-    );
+    var config = sdk.newSpaceSettingsBuilder();
+    config.setName(spaceName);
+    if (description != null) {
+      config.setTopic(description);
+    }
+    if (avatarUri.isNotEmpty) {
+      config.setAvatarUri(avatarUri);
+    }
```
[the remark this answers is not in the document]
Comment (bitfriend) [reply app/lib/features/spaces/dialogs/create_space_sheet.dart]: `CreateSpaceSettings` contains `avatar_uri` field.
```rs
#[derive(Builder, Default, Deserialize, Serialize, Clone)]
pub struct CreateSpaceSettings {
    #[builder(setter(strip_option))]
    name: Option<String>,

    #[builder(default = "Visibility::Private")]
    #[serde(default = "space_visibilty_default")]
    visibility: Visibility,

    #[builder(default = "Vec::new()")]
    #[serde(default)]
    invites: Vec<OwnedUserId>,

    #[builder(setter(strip_option), default)]
    alias: Option<String>,

    #[builder(setter(strip_option), default)]
    topic: Option<String>,

    #[builder(setter(strip_option), default)]
    avatar_uri: Option<String>,

    #[builder(setter(strip_option), default)]
    parent: Option<OwnedRoomId>,
}
```

File: native/core/src/spaces.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (bitfriend) [reply native/core/src/spaces.rs]: Yeah, just split `set_invites()` into `add_invitee()`.

**Comment by bitfriend** [comment]
Now user can specify remote uri or local uri for convo avatar and space avatar.

File: app/lib/features/spaces/dialogs/create_space_sheet.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (bitfriend) [reply app/lib/features/spaces/dialogs/create_space_sheet.dart]: Fixed.
Now user can specify avatar when creating space.

File: native/acter/api.rsh (hunk 3 above)
[the remark this answers is not in the document]
Comment (bitfriend) [reply native/acter/api.rsh:1370]: Fixed

**Review by bitfriend — COMMENTED** [review]
Nice work

**Pull request merged**